src/workerd/api/node/tests/http-server-nodejs-test.js

```diff
@@ -184,6 +193,124 @@ export const testHttpServerMultiHeaders = {
   },
 };
 
+// Test is taken from test/parallel/test-http-server-multiheaders2.js
+export const testHttpServerMultiHeaders2 = {
+  async test(_ctrl, env) {
+    // One difference between Node.js and Cloudflare workers is that Cookie is allowed
+    // to have multiple values but in Workers it is not supported.
+    const multipleAllowed = [
+      'Accept',
+      'Accept-Charset',
+      'Accept-Encoding',
+      'Accept-Language',
+      'Connection',
+      'DAV', // GH-2750
+      'Pragma', // GH-715
+      'Link', // GH-1187
+      'WWW-Authenticate', // GH-1083
+      'Proxy-Authenticate', // GH-4052
+      'Sec-Websocket-Extensions', // GH-2764
+      'Sec-Websocket-Protocol', // GH-2764
+      'Via', // GH-6660
+
+      // not a special case, just making sure it's parsed correctly
+      'X-Forwarded-For',
+
+      // Make sure that unspecified headers is treated as multiple
+      'Some-Random-Header',
+      'X-Some-Random-Header',
+    ];
+
+    const multipleForbidden = [
+      'Content-Type',
+      'User-Agent',
+      'Referer',
+      'Host',
+      'Authorization',
+      'Proxy-Authorization',
+      'If-Modified-Since',
+      'If-Unmodified-Since',
+      'From',
+      'Location',
+      'Max-Forwards',
+    ];
+
+    await using server = http.createServer(function (req, res) {
+      for (const header of multipleForbidden) {
+        const value = req.headers[header.toLowerCase()];
+        strictEqual(
+          value,
+          'foo',
+          `multiple forbidden header parsed incorrectly: ${header} with value: "${value}"`
+        );
+      }
+      for (const header of multipleAllowed) {
+        const sep = header.toLowerCase() === 'cookie' ? '; ' : ', ';
+        strictEqual(
+          req.headers[header.toLowerCase()],
+          `foo${sep}bar`,
+          `multiple allowed header parsed incorrectly: ${header}`
+        );
+      }
+
+      res.writeHead(200, { 'Content-Type': 'text/plain' });
+      res.end('EOF');
+    });
+
+    function makeHeader(value) {
+      return function (header) {
+        return [header, value];
+      };
+    }
+
+    const headers = []
+      .concat(multipleAllowed.map(makeHeader('foo')))
+      .concat(multipleForbidden.map(makeHeader('foo')))
+      .concat(multipleAllowed.map(makeHeader('bar')))
+      .concat(multipleForbidden.map(makeHeader('bar')));
+
+    server.listen(8080);
+
+    const res = await env.SERVICE.fetch('https://cloudflare.com/', {
+      headers,
+    });
+    strictEqual(res.status, 200);
+  },
+};
+
+// Test for RFC 7230 compliant header splitting with quoted strings
+export const testHttpServerQuotedStringHeaders = {
+  async test(_ctrl, env) {
+    await using server = http.createServer((req, res) => {
+      strictEqual(req.headers['content-type'], 'text/plain; f="a, b, c"');
+      strictEqual(req.headers['authorization'], 'Bearer token="abc, def"');
```

Comment:
Perhaps include an escaped char (e.g. `\"') in the quoted portion of these to ensure that those are being handled correctly